server/reportConfiguration/analyseNomisReportConfiguration.ts

```diff
@@ -55,3 +72,37 @@ export function analyseNomisReportConfiguration(incidentTypes: readonly Incident
 
   return results
 }
+
+function enumeratePaths(firstQuestionId: number, questionConnections: Map<number, Set<number>>): number {
+  function walkPath(paths: number[][], path: number[], nextQuestionId: number | null) {
+    if (nextQuestionId === null || nextQuestionId === undefined) {
+      return
+    }
+    const nextQuestionIds = questionConnections.get(nextQuestionId)
+    if (!nextQuestionIds) {
+      throw new Error(`Question id ${nextQuestionId} is not within report`)
+    }
+    if (path.includes(nextQuestionId)) {
+      throw new Error(`Question cycle detected: ${path} loops back to ${nextQuestionId}`)
+    }
+    path.push(nextQuestionId)
+    const clonedPath = structuredClone(path)
```

Comment:
I can't remember if I mentioned this to you....but I ran into problems with `structuredClone()` early on with one of the bigger report type config.

Surprisingly it was using a lot of memory, not sure if it was a bug in my logic - possible - but it wasn't happening when traversing other report configs. Something to bear in mind.

Also, I think you could use [the `Graph` data structure](https://github.com/ministryofjustice/hmpps-incident-reporting/blob/main/server/data/incidentTypeConfiguration/directedGraphs.ts) here to check for cycles and other problems like I do when I validate the DPS config.

You'd need to do [something similar to this](https://github.com/ministryofjustice/hmpps-incident-reporting/blob/main/server/data/incidentTypeConfiguration/validation.ts#L53-L65) but to generate the graph from the NOMIS config, then call `dfs()` to do the traversal and you can use the result to determine if there are cycles or unreachable nodes, etc...